plugins/modules/credential.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+
+# (c) 2023, Nikhil Jain <nikjain@redhat.com>
+# (c) 2024, Alina Buzachis <abuzachis@redhat.com>
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = """
+---
+module: credential
+author:
+   - "Nikhil Jain (@jainnikhil30)"
+   - "Alina Buzachis (@alinabuzachis)"
+short_description: Manage credentials in EDA Controller
+description:
+    - This module allows you to create, update or delete a credential in EDA controller.
+version_added: 2.0.0
+options:
+  name:
+    description:
+      - Name of the credential.
+    type: str
+    required: true
+  new_name:
+    description:
+      - Setting this option will change the existing name (lookup via name).
+    type: str
+  inputs:
+    description:
+      - Credential inputs where the keys are var names used in templating.
+    type: dict
+  credential_type_id:
+    description:
+      - The ID of the credential type.
+    type: int
+  organization_id:
+    description:
+      - The ID of the organization.
+    type: int
+    aliases:
+      - org_id
+  description:
+    description:
+      - Description of the credential.
+    type: str
+  state:
+    description:
+      - Desired state of the resource.
+    default: "present"
+    choices: ["present", "absent", "exists"]
```

Comment:
```suggestion
    choices: ["present", "absent"]
```

I would drop `exists` state since it is similar to `present`.